db/db_iter.cc

```diff
@@ -1255,8 +1257,9 @@ void DBIter::SeekToFirst() {
   } else {
     valid_ = false;
   }
-  if (valid_ && prefix_same_as_start_) {
-    assert(prefix_extractor_ != nullptr);
+  assert(!prefix_same_as_start_ || prefix_extractor_ != nullptr);
+  if (valid_ && prefix_same_as_start_ &&
+      prefix_extractor_->InDomain(saved_key_.GetUserKey())) {
```

Comment:
If `valid_`, can prefix_extractor_->InDomain(saved_key_.GetUserKey())` be false?